app/components/AddressBlock.js

```diff
@@ -178,18 +176,17 @@ type State = {
 class AddressBlock extends Component<Props, State> {
   props: Props;
   state = {
-    shouldHideSmartAddressesInfo: false
+    shouldHideSmartAddressesInfo: false,
+    isDelegateModalOpen: false
   };
 
-  handleManagerAddressClick = () => {
-    const { accountBlock } = this.props;
-    const publicKeyHash = accountBlock.get('publicKeyHash');
-
-    this.selectAccount(publicKeyHash, publicKeyHash);
-  };
+  openDelegateModal = () =>  this.setState({ isDelegateModalOpen: true });
+  closeDelegateModal = () =>  this.setState({ isDelegateModalOpen: false });
 
-  selectAccount = (accountHash: string, parentHash: string) => {
-    this.props.selectAccount(accountHash, parentHash);
+  goToAccount = (selectedAccountHash, selectedParentHash) => {
```

Comment:
Do you think `goToAccount` will be used in other places? If so then you could put it in the store and you could just route from the store with `react-router-redux`